src/checkout/Dockerfile

```
@@ -2,25 +2,27 @@
 # SPDX-License-Identifier: Apache-2.0
 
 
-FROM golang:1.22-alpine AS builder
+FROM golang:1.24-bookworm AS builder
 
 WORKDIR /usr/src/app/
 
-RUN --mount=type=cache,target=/go/pkg/mod/ \
-    --mount=type=bind,source=./src/checkout/go.sum,target=go.sum \
-    --mount=type=bind,source=./src/checkout/go.mod,target=go.mod \
-    go mod download
+COPY ./src/checkout/go.mod go.mod
+COPY ./src/checkout/go.sum go.sum
 
-RUN --mount=type=cache,target=/go/pkg/mod/ \
-    --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=bind,rw,source=./src/checkout,target=. \
-    go build -ldflags "-s -w" -o /go/bin/checkout/ ./
+RUN go mod download
 
-FROM alpine
+COPY ./src/checkout/genproto/oteldemo/ genproto/oteldemo/
+COPY ./src/checkout/kafka/ kafka/
+COPY ./src/checkout/money/ money/
+COPY ./src/checkout/main.go main.go
+
+RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -ldflags "-s -w" -o checkout main.go
```

```suggestion
RUN CGO_ENABLED=0 GOOS=linux go build -ldflags "-s -w" -o checkout main.go
```

My understanding is that GO111MODULE is ignored since Go 1.17.